### ISSUE TITLE
agent container stops working

### ISSUE THREAD

**Issue by calohmn** [EnMasseProject/enmasse#1052]
We just had a scenario, where at some point, newly created addresses didn't get registered in qdrouter and broker any more.
When checking the address status, the output was e.g.
````
"Address event/tst_9edb2f8978714d89b6f6562ef504fb59 is missing autoLinks on qdrouterd-5cf8d6cf46-qwrcg",
"Address event/tst_9edb2f8978714d89b6f6562ef504fb59 not found on qdrouterd-5cf8d6cf46-qwrcg",
"Address event/tst_9edb2f8978714d89b6f6562ef504fb59 is missing active autoLink (active in dirs: [])"
````

Looking at the log of the agent container, the messages `syncing broker` or `updating addresses` stopped appearing at some point:
````
2018-03-13T11:01:20.583Z agent info updating addresses for Router.qdrouterd-5cf8d6cf46-qwrcg
2018-03-13T11:01:47.188Z agent info connection 100.119.85.199:37128 has been deleted, notifying clients...
2018-03-13T11:02:18.680Z agent info response ended; reconnecting...
2018-03-13T11:02:18.696Z agent info GET /api/v1/namespaces/hono/pods?labelSelector=role%3Dbroker => 200 
2018-03-13T11:02:18.697Z agent info list retrieved; watching...
2018-03-13T11:02:18.701Z agent info GET /api/v1/watch/namespaces/hono/pods?labelSelector=role%3Dbroker => 200 
2018-03-13T11:02:52.190Z agent info connection 100.119.85.199:37468 has been deleted, notifying clients...
2018-03-13T11:03:57.197Z agent info connection 100.119.85.199:37794 has been deleted, notifying clients...
2018-03-13T11:05:02.205Z agent info connection 100.119.85.199:38088 has been deleted, notifying clients...
2018-03-13T11:06:02.211Z agent info connection 100.119.85.199:38394 has been deleted, notifying clients...
2018-03-13T11:07:07.221Z agent info connection 100.119.85.199:38710 has been deleted, notifying clients...
2018-03-13T11:08:12.234Z agent info connection 100.119.85.199:39008 has been deleted, notifying clients...
2018-03-13T11:09:12.237Z agent info connection 100.119.85.199:39332 has been deleted, notifying clients...
2018-03-13T11:10:17.246Z agent info connection 100.119.85.199:39660 has been deleted, notifying clients...
2018-03-13T11:10:48.623Z agent info response ended; reconnecting...
2018-03-13T11:10:48.634Z agent info response ended; reconnecting...
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 55): Error: connect ECONNREFUSED 100.64.0.1:443
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 56): Error: connect ECONNREFUSED 100.64.0.1:443
2018-03-13T11:11:27.257Z agent info connection 100.119.85.199:40012 has been deleted, notifying clients...
2018-03-13T11:11:42.256Z agent info connection 100.119.85.199:40096 has been deleted, notifying clients...
2018-03-13T11:12:42.265Z agent info connection 100.119.85.199:40436 has been deleted, notifying clients...
2018-03-13T11:13:02.274Z agent info connection 100.119.85.199:40504 has been deleted, notifying clients...
2018-03-13T11:14:02.286Z agent info connection 100.119.85.199:40854 has been deleted, notifying clients...
2018-03-13T11:14:22.278Z agent info connection 100.119.85.199:40968 has been deleted, notifying clients...
2018-03-13T11:15:02.293Z agent info connection 100.119.85.199:32922 has been deleted, notifying clients...
2018-03-13T11:15:02.293Z agent info connection 100.119.85.199:32924 has been deleted, notifying clients...
2018-03-13T11:15:02.293Z agent info connection 100.119.85.11:41518 has been deleted, notifying clients...
2018-03-13T11:16:12.297Z agent info connection 100.119.85.199:41684 has been deleted, notifying clients...
2018-03-13T11:17:17.303Z agent info connection 100.119.85.199:42502 has been deleted, notifying clients...
2018-03-13T11:17:22.307Z agent info connection 100.119.85.199:42572 has been deleted, notifying clients...
2018-03-13T11:18:22.328Z agent info connection 100.119.85.199:43218 has been deleted, notifying clients...
2018-03-13T11:19:27.460Z agent info connection 100.119.85.199:43920 has been deleted, notifying clients...
2018-03-13T11:19:37.333Z agent info connection 100.119.85.199:41244 has been deleted, notifying clients...
2018-03-13T11:19:37.333Z agent info connection 100.119.85.199:41246 has been deleted, notifying clients...
2018-03-13T11:21:47.357Z agent info connection 100.119.87.9:52668 has been deleted, notifying clients...
2018-03-13T11:22:52.368Z agent info connection 100.119.87.9:53252 has been deleted, notifying clients...
2018-03-13T11:23:57.377Z agent info connection 100.119.87.9:53858 has been deleted, notifying clients...
2018-03-13T11:25:02.433Z agent info connection 100.119.87.9:54416 has been deleted, notifying clients...
2018-03-13T11:26:07.400Z agent info connection 100.119.87.9:54988 has been deleted, notifying clients...
2018-03-13T11:27:12.416Z agent info connection 100.119.87.9:55594 has been deleted, notifying clients...
2018-03-13T11:28:12.422Z agent info connection 100.119.87.9:56170 has been deleted, notifying clients...
2018-03-13T11:29:17.432Z agent info connection 100.119.87.9:56768 has been deleted, notifying clients...
2018-03-13T11:30:17.492Z agent info connection 100.119.87.9:57334 has been deleted, notifying clients...
2018-03-13T11:31:22.453Z agent info connection 100.119.87.9:57890 has been deleted, notifying clients...
2018-03-13T11:32:27.462Z agent info connection 100.119.87.9:58456 has been deleted, notifying clients...
2018-03-13T11:33:27.479Z agent info connection 100.119.87.9:59042 has been deleted, notifying clients...
2018-03-13T11:34:32.480Z agent info connection 100.119.87.9:59610 has been deleted, notifying clients...
2018-03-13T11:35:37.499Z agent info connection 100.119.87.9:60206 has been deleted, notifying clients...
2018-03-13T11:36:42.496Z agent info connection 100.119.87.9:60772 has been deleted, notifying clients...
[connection-15] error: Error: read ECONNRESET
[connection-16] error: Error: read ECONNRESET
[connection-43] error: Error: read ECONNRESET
[connection-45] error: Error: read ECONNRESET
2018-03-13T11:37:27.640Z agent info Accepted incoming websocket connection
2018-03-13T11:37:27.999Z agent info GET /api/v1/namespaces/hono/configmaps?labelSelector=type%3Daddress-plan => 200 
[connection-52] error: Error: read ECONNRESET
[connection-54] error: Error: read ECONNRESET
[connection-55] error: Error: read ECONNRESET
2018-03-13T11:37:47.506Z agent info connection 100.119.87.9:33132 has been deleted, notifying clients...
2018-03-13T11:38:52.511Z agent info connection 100.119.87.9:33722 has been deleted, notifying clients...
2018-03-13T11:39:57.515Z agent info connection 100.119.87.9:34328 has been deleted, notifying clients...
2018-03-13T11:41:02.518Z agent info connection 100.119.87.9:34884 has been deleted, notifying clients...
2018-03-13T11:42:02.524Z agent info connection 100.119.87.9:35458 has been deleted, notifying clients...
2018-03-13T11:43:07.540Z agent info connection 100.119.87.9:36046 has been deleted, notifying clients...
2018-03-13T11:44:12.551Z agent info connection 100.119.87.9:36616 has been deleted, notifying clients...
2018-03-13T11:45:17.567Z agent info connection 100.119.87.9:37178 has been deleted, notifying clients...
2018-03-13T11:46:17.579Z agent info connection 100.119.87.9:37758 has been deleted, notifying clients...
2018-03-13T11:47:22.593Z agent info connection 100.119.87.9:38318 has been deleted, notifying clients...
2018-03-13T11:48:27.610Z agent info connection 100.119.87.9:38904 has been deleted, notifying clients...
2018-03-13T11:49:27.618Z agent info connection 100.119.87.9:39496 has been deleted, notifying clients...
2018-03-13T11:50:32.632Z agent info connection 100.119.87.9:40048 has been deleted, notifying clients...
2018-03-13T11:51:32.649Z agent info connection 100.119.87.9:40612 has been deleted, notifying clients...
2018-03-13T11:52:37.660Z agent info connection 100.119.87.9:41180 has been deleted, notifying clients...
````
So, I guess, connections got lost and didn't get re-established for some reason.

After restarting the admin pod, the created addresses got ready.

Used version is 0.17.1.

**Comment by grs** [comment]
Do you have the full log (or even any more of it)?

**Comment by calohmn** [comment]
Full logs are here: [logs.zip](https://github.com/EnMasseProject/enmasse/files/1807311/logs.zip).

**Comment by grs** [comment]
I can't say for sure, but I think the fix committed in PR #1063 may fix this. Certainly it fixes a bug that would give similar symptoms. I suggest we close this and re-open if it occurs again?

**Comment by calohmn** [comment]
Ok, good.
